unity-renderer/Assets/Scripts/Example.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using UnityEngine;
+
+namespace DefaultNamespace
```

Comment:
we should also have a standard for namespaces. I think they should follow the folder structure in the best case scenario (as Rider suggests), but in DCL we do have a mess in that regard which implies a very big refactor on the folder structure. For example:
- Folder: `DCL/Social/HUD`
- Namespace: `DCL.Social.HUD`

To mitigate this issue, we are setting a root namespace for the feature, ignoring the folder structure. For example:
- Folder: `/Scripts/MainScripts/DCL/Controllers/ChatController/`
- Namespace: `DCL.Social.Chat`

If you do have a nested folder then:
- Folder: `/Scripts/MainScripts/DCL/Controllers/ChatController/InnerGroup`
- Namespace: `DCL.Social.Chat.InnerGroup`

Reply:
@Antivortex , it is very easy In Rider  - right-click on the folder in the Explorer window (or Alt+Enter) and then disabling `Namespace provider` checkbox
![image](https://user-images.githubusercontent.com/35366872/191733898-77e76b85-ba45-4231-b705-2a9e7d21bb95.png)

Reply:
I think this doesn't work really well with Unity's asmdef because they are forced into different folders. It will be even more difficult once we merge desktop/webgl repo and become truly multiplatform. How would you deal with interfaces?

##  Case A

Let's say you have:
- `interface IMyService` 
- `class MyService_WebGL : IMyService`
- `class MyService_Desktop : IMyService`

I guess we can solve it with paths like:
- `Assets/Scripts/Common/MyService/IMyService.cs`
- `Assets/Scripts/WebGL/MyService/MyService_WebGL.cs`
- `Assets/Scripts/Desktop/MyService/MyService_Desktop.cs`

and we would strip `Assets/Scripts/****/`


## Case B
You have:
- `class ServiceA : IServiceA`
- `class SystemB`  (which uses `IServiceA`)

You want to split `ServiceA` into a _definitions_ and _implementation_ folders with different assemblies to avoid cycling references. Paths would be like
```
- ServiceA/
---- ServiceA.asmdef
---- ServiceA.cs
---- Definitions/
-------- ServiceADefinitions.asmdef
-------- IServiceA.cs
```

I would find really ugly to have a namespace `ServiceA.Definitions` to access `IServiceA`. How could we deal with this.

Reply:
It is a really good point and complex topic about namespaces vs asmdefs vs folder structure. I suggest discussing it separately (as the third iteration of refinement) - first namings, second order, and third namespaces. So let's take these arguments to later discussion.

Reply:
One solution for the problem **Case B** mentioned by @AjimenezDCL, would be to make namespaces wider and not present them for small folders (so to not go very detailed). In Case B, I would propose to not present `ServiceA.Definitions` namespace and only have the namespace `ServiceA` for both definitions and implementation. 
In the end, the main goal of namespaces is to avoid names collisions